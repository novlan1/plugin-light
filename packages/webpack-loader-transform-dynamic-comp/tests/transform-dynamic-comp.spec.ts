import path from 'path';
import transformDynamicComp from '../src/loader';


beforeAll(() => {
  process.env.UNI_PLATFORM = 'mp-weixin';
});


describe('transformDynamicComp', () => {
  it('transformDynamicComp', () => {
    const source = `
<script>
export default {
  components: {
    xxComp(resolve) {
      require(['xx.comp'], resolve);
    },
  }
}
</script>
    `;
    expect(transformDynamicComp.call(
      { resourcePath: path.resolve(process.cwd(), './src', 'MOCK_PAGE.vue') },
      source,
    )).toMatchSnapshot();
  });
});
