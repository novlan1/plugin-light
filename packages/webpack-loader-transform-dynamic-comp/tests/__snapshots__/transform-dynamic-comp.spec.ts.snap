// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformDynamicComp transformDynamicComp 1`] = `
"
<script>
import xxComp from 'xx.comp';

export default {
  components: {
    xxComp,
  }
}
</script>
    "
`;
